sunpy/net/fido_factory.py

```diff
@@ -43,25 +44,25 @@ def __init__(self, *results):
         """
         Parameters
         ----------
-        results : `sunpy.net.base_client.BaseQueryResponse`
+        results : `sunpy.net.base_client.QueryResponseTable`
             One or more QueryResponse objects.
         """
         self._list = []
         self._numfile = 0
         for result in results:
-            if not isinstance(result, BaseQueryResponse):
+            if isinstance(result, QueryResponseRow):
+                result = result.as_table()
+
+            if not isinstance(result, QueryResponseTable):
                 raise TypeError(
-                    f"{type(result)} is not derived from sunpy.net.base_client.BaseQueryResponse")
+                    f"{type(result)} is not derived from sunpy.net.base_client.QueryResponseTable")
 
             self._list.append(result)
-            self._numfile = len(result)
+            self._numfile += len(result)
```

Comment:
This seems like a potentially major bug that you've fixed?

Reply:
yeah, it does...